### PR TITLE
EWPP-176: Add concept subset for countries vocabulary.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Add a concept subset plugin to filter the countries vocabulary

### Change log

- Added: Corporate countries subset.
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/Plugin/ConceptSubset/DeprecatedCountries.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+use Drupal\rdf_entity\RdfFieldHandlerInterface;
+use Drupal\rdf_skos\ConceptSubsetPluginBase;
+use Drupal\rdf_skos\Plugin\PredicateMapperInterface;
+
+/**
+ * Creates a subset of the countries vocabulary.
+ *
+ * @ConceptSubset(
+ *   id = "deprecated_countries",
+ *   label = @Translation("Deprecated countries"),
```
Comment (brummbar) [inline src/Plugin/ConceptSubset/DeprecatedCountries.php:18]:
```suggestion
 *   label = @Translation("Non-deprecated countries"),
```

File: src/Plugin/ConceptSubset/DeprecatedCountries.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\rdf_entity\RdfFieldHandlerInterface;
+use Drupal\rdf_skos\ConceptSubsetPluginBase;
+use Drupal\rdf_skos\Plugin\PredicateMapperInterface;
+
+/**
+ * Creates a subset of the countries vocabulary.
+ *
+ * @ConceptSubset(
+ *   id = "deprecated_countries",
```
Comment (brummbar) [inline src/Plugin/ConceptSubset/DeprecatedCountries.php:17]:
```suggestion
 *   id = "non_deprecated_countries",
```

File: tests/src/Kernel/CorporateCountrySubsetTest.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    $reference_field->save();
+
+    // Referencing a deprecated country triggers a validation error.
+    /** @var \Drupal\entity_test\Entity\EntityTest $entity */
+    $entity = $entity_type_manager->getStorage('entity_test')
+      ->create(['type' => 'entity_test']);
+    $entity->set('field_country', 'http://publications.europa.eu/resource/authority/country/ANT');
+    $violations = $entity->field_country->validate();
+    $this->assertCount(1, $violations);
+    $this->assertEquals(t('This entity (%type: %id) cannot be referenced.', ['%type' => 'skos_concept', '%id' => 'http://publications.europa.eu/resource/authority/country/ANT']), $violations[0]->getMessage());
```
Comment (brummbar) [inline tests/src/Kernel/CorporateCountrySubsetTest.php:68]: Let's not use the t() in tests, we know we are looking for an english text. Use (string) on the violation message.

File: src/Plugin/ConceptSubset/DeprecatedCountries.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+namespace Drupal\oe_corporate_countries\Plugin\ConceptSubset;
+
+use Drupal\Core\Entity\Query\QueryInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\rdf_entity\RdfFieldHandlerInterface;
+use Drupal\rdf_skos\ConceptSubsetPluginBase;
+use Drupal\rdf_skos\Plugin\PredicateMapperInterface;
+
+/**
+ * Creates a subset of the countries vocabulary.
```
Comment (brummbar) [inline src/Plugin/ConceptSubset/DeprecatedCountries.php:14]:
```suggestion
 * Subset of the countries vocabulary with non-deprecated countries only.
```

File: src/Plugin/ConceptSubset/DeprecatedCountries.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ *   }
+ * )
+ */
+class DeprecatedCountries extends ConceptSubsetPluginBase implements PredicateMapperInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alterQuery(QueryInterface $query, $match_operator, array $concept_schemes = [], string $match = NULL): void {
+    $query->condition('deprecated', 'false');
```
Comment (brummbar) [inline src/Plugin/ConceptSubset/DeprecatedCountries.php:32]: I used a check in the country repository for true or 1. Maybe we should create a condition and put both false or 0, but I don't see any in the vocabulary. Let's go like it is now.

**Review by ademarco — DISMISSED** [review]
Manually tested on contact forms, it works well.

**Review by brummbar — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**